pkg/gitversion/convert.go

```diff
@@ -65,9 +65,11 @@ func getPythonPreVersion(preVersion string) (string, error) {
 	}
 
 	// Find the hash and remove to make sure we don't pick up the hash as a number
-	hashRe := regexp.MustCompile(`[0-9a-f]{8}`)
-	shortHash := hashRe.FindString(remaining)
-	if shortHash != "" {
+	hashRe := regexp.MustCompile(`\b[0-9a-f]{8}\b`)
```

Comment:
Should we test for "\+[0-9a-f]{8}" here instead? I don't think the "+" can show up anywhere else in a semver? Then you can change to only expecting to find one or zero below

Reply:
I think it can show up differently if combined with +dirty

Reply:
Or maybe it's always .dirty 🤔